api/src/main/java/org/pmiops/workbench/compliance/ComplianceServiceImpl.java

```diff
@@ -21,18 +23,35 @@
   private static final String MOODLE_EXCEPTION = "moodle_exception";
   private static final String MOODLE_USER_NOT_ALLOWED_ERROR_CODE = "guestsarenotallowed";
   private CloudStorageService cloudStorageService;
+  private Provider<WorkbenchConfig> configProvider;
+
 
   @Autowired
-  public ComplianceServiceImpl(CloudStorageService cloudStorageService) {
+  public ComplianceServiceImpl(CloudStorageService cloudStorageService,
+      Provider<WorkbenchConfig> configProvider) {
     this.cloudStorageService = cloudStorageService;
+    this.configProvider = configProvider;
   }
 
   private String getToken() {
     return this.cloudStorageService.getMoodleApiKey();
   }
 
+  private boolean enableMoodleCalls() {
+    return configProvider.get().moodle.enableMoodleBackend;
+  }
+
+  private void setApiHost() {
+    ApiClient apiClient = api.getApiClient();
+    apiClient.setBasePath("https://" + configProvider.get().moodle.host +"/webservice/rest");
+  }
+
   @Override
   public Integer getMoodleId(String email) throws ApiException {
+    if (!enableMoodleCalls())
+      return null;
+    setApiHost();
+    api.getApiClient().setDebugging(true);
```

Comment:
I'm not sure what the impact of turning on API client debugging is – should this be on in production or was it just for local testing?

Reply:
I forgot to remove this, it was just for my testing purpose